mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -900,6 +903,25 @@ private static DBObject stage(String key, Object value) {
 		return stage;
 	}
 
+	private static ClosableIterator<Tuple> doDistinct(
+			final MongoDBQueryDescriptor queryDescriptor, final DBCollection collection) {
+		List distinctFieldValues = collection.distinct( queryDescriptor.getFieldName(), queryDescriptor.getCriteria() );
```

Comment:
The tuple context is not available, probably because we are dealing with more than a single entity type.